src/GraphQL.AotCompilationSample/Program.cs

```diff
@@ -0,0 +1,180 @@
+// See https://aka.ms/new-console-template for more information
+using System.Diagnostics.CodeAnalysis;
+using GraphQL;
+using GraphQL.StarWars;
+using GraphQL.Types;
+using Microsoft.Extensions.DependencyInjection;
+
+Console.WriteLine("Sample of AOT compilation of a GraphQL query");
+Console.WriteLine();
+
+IServiceCollection serviceCollection = new ServiceCollection();
+// build services as usual; most methods are fully supported except:
+//   - AddClrTypeMappings
+//   - AddAutoClrMappings
+//   - AddAutoSchema
+serviceCollection.AddGraphQL(b => b
+    .AddSystemTextJson()
+    //.AddSchema<StarWarsSchema>()
+    //.AddGraphTypes(typeof(StarWarsSchema).Assembly)
+    .AddSelfActivatingSchema<StarWarsSchema>()
+);
+
+serviceCollection.AddSingleton<StarWarsData>();
+
+// must manually preserve the query and mutation types or AOT will trim their constructors
+// all other graph types' constructors are preserved via calls to Field<T>
+Preserve<StarWarsQuery>();
+Preserve<StarWarsMutation>();
+
+// other notes:
+// - auto clr type mappings are generally not supported
+// - auto registering graph types are generally not supported
+// - field builders that use Expressions as resolvers such as Field(x => x.Name) are discouraged
+// - field builders that do not include a resolver such as Field<StringGraphType>("Name") are not supported
+// - strongly recommend each field has the explicit graph type specified and a resolver specified
+
+#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+var services = serviceCollection.BuildServiceProvider();
+#pragma warning restore IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
+
+var executer = services.GetRequiredService<IDocumentExecuter>();
+
+var ret = await executer.ExecuteAsync(new ExecutionOptions
+{
+    Schema = services.GetRequiredService<ISchema>(),
+    Query = "{ hero { id name } }",
+    RequestServices = services,
+    ThrowOnUnhandledException = true,
+}).ConfigureAwait(false);
+
+var serializer = services.GetRequiredService<IGraphQLTextSerializer>();
+var response = serializer.Serialize(ret);
+
+Console.WriteLine("Executing request: { hero { id name } }");
+Console.WriteLine();
+Console.WriteLine(response);
+Console.WriteLine();
+
+if (response != """{"data":{"hero":{"id":"3","name":"R2-D2"}}}""")
+{
+    Console.WriteLine("Unexpected response; exiting.");
+    return 1; // return application exit code of 1 indicating failure
+}
+
+var introspectionQuery = """
```

Comment:
I would like to move this query into separate file as "content" to make example more minimalistic.

Reply:
Sure.